libraries/AP_Proximity/AP_Proximity_TeraRangerTower.cpp

```diff
@@ -91,12 +91,15 @@ bool AP_Proximity_TeraRangerTower::read_sensor_data()
 
 // process reply
 void AP_Proximity_TeraRangerTower::update_sector_data(int16_t angle_deg, uint16_t distance_cm)
-{
-    const uint8_t sector = convert_angle_to_sector(angle_deg);
-    _angle[sector] = angle_deg;
-    _distance[sector] = ((float) distance_cm) / 1000;
-    _distance_valid[sector] = distance_cm != 0xffff;
+{   
+    // Get location on 3-D boundary based on angle to the object
+    const AP_Proximity_Boundary_3D::Face face = boundary.get_face(angle_deg);
+    if (distance_cm != 0xffff) {
+        boundary.set_face_attributes(face, angle_deg, ((float) distance_cm) / 1000);
```

Comment:
One has to worry about this code a little.....
cm / 1000 is 10s-of-meters....

Please tell me we should rename the parameter here....

Similarly in the evo

Reply:
Randy and I have decided to keep it as it is, till we can actually test it with a real sensor. Although we do agree something needs to be done about it.